### ISSUE TITLE
Update Cython integration 

### ISSUE THREAD

**Issue by LinguList** [lingpy/lingpy#295]
Apparently, there is a preferred way how to do cython integration if one wants to maintain pure python code:

* http://docs.cython.org/en/latest/src/tutorial/pure.html

This seems to be straightforward, but a bit tedious, yes for the alignment parts, it may be useful to switch to this implementation.

**Comment by xrotwang** [comment]
Yes, this looks like we could get rid of https://github.com/lingpy/lingpy/blob/master/lingpy/algorithm/cython/compilePYX.py

**Comment by LinguList** [comment]
Just checked this, and it basically works. The only tedious aspect is to get all definitions into the *.pxd file, which is mostly just boring, but luckily, I put all definitions up in front. The way to do it is to write a *.pxd file:

```python
import cython

# we start with basic alignment functions
@cython.locals( i=cython.int, j=cython.int, almA=cython.list, almB=cython.list,
        matrix=cython.list, traceback=cython.list, gapA=cython.float,
        gapB=cython.float, match=cython.float, sim=cython.float)
cpdef tuple globalign( list seqA, list seqB, list gopA, list gopB, str proA, str
        proB, int M, int N, float scale, float factor, object scorer)
```
and then have the normal *.py file with the full function:

```python
def globalign(
        seqA,
        seqB,
        gopA,
        gopB,
        proA,
        proB,
        M, # length of seqA
        N, # length of seqB
        scale,
        factor,
        scorer
        ):
    """
    Carry out global alignment of two sequences.
...
```

Compiling to *.c is then done with cython:

```shell
$ cython file.py
```
And I checked: if you don't use the *.pxd file, the c-code will declare all data as python objects, which is where we use speed.

As we only submit the C-files anyway, it seems like this is the best way to go in the future, probably also for lingpy3, as the alignment algorithms are (to my best ability) written very carefully with an eye on speed, and I would not like to loose the potential. 

But I'd say that this should be done for lingpy2.7, 2.6 is too early.
